controller/space.go

```diff
@@ -66,18 +84,38 @@ func (c *SpaceController) Create(ctx *app.CreateSpaceContext) error {
 		*/
 
 		newArea := area.Area{
-			ID:      satoriuuid.NewV4(),
-			SpaceID: space.ID,
-			Name:    space.Name,
+			ID:      uuid.NewV4(),
+			SpaceID: rSpace.ID,
+			Name:    rSpace.Name,
 		}
 		err = appl.Areas().Create(ctx, &newArea)
 		if err != nil {
-			return jsonapi.JSONErrorResponse(ctx, errs.Wrapf(err, "failed to create area: %s", space.Name))
+			return jsonapi.JSONErrorResponse(ctx, errs.Wrapf(err, "failed to create area: %s", rSpace.Name))
 		}
 
 		res := &app.SpaceSingle{
-			Data: ConvertSpace(ctx.RequestData, space),
+			Data: ConvertSpace(ctx.RequestData, rSpace),
+		}
+
+		// Create keycloak resource for this space
+		resource, err := c.resourceManager.CreateResource(ctx, ctx.RequestData, rSpace.ID.String(), spaceResourceType, &rSpace.Name, &scopes, rSpace.OwnerId.String(), rSpace.Name+"-"+uuid.NewV4().String())
```

Comment:
The only reason is for not creating the remote resource before would be that we need to know the space ID. But it's actually an issue here because the space ID is just a uuid so we can generate it first. But then we have the same problem with a created but unused KC resource in case the space transaction fails. Which is not critical.
So,
1. Generate space uuid.
2. Create KC resource
3. Transaction for creating space & space resource in DB.
4. If transaction fails then remove the KC resource 

Or there is something better we can do?